sound/soc/sof/xtensa/core.c

```diff
@@ -132,6 +132,17 @@ static void xtensa_stack(struct snd_sof_dev *sdev, const char *level, void *oops
 				   buf, sizeof(buf), false);
 		dev_printk(level, sdev->dev, "0x%08x: %s\n", stack_ptr + i * 4, buf);
 	}
+
+	if (!xoops->plat_hdr.numaregs)
+		return;
+
+	dev_printk(level, sdev->dev, "AR registers:\n");
+	/* the number of ar registers is a multiple of 4 */
+	for (i = 0; i < xoops->plat_hdr.numaregs && i < XTENSA_CORE_MAX_AR_REGS_COUNT; i += 4) {
```

Comment:
well, if the `xoops->plat_hdr.numaregs` is invalid then likely the whole dump is corrupted..
Printing a known corrupted dump have questionable benefit.

and the line can be shortened if it is really desired to print even if the numregs is out of range:
```c
	u32 num_regs;
	...
	num_regs = min(xoops->plat_hdr.numaregs, XTENSA_CORE_MAX_AR_REGS_COUNT);
	for (i = 0; i < num_regs; i += 4) {
```

Reply:
@ujfalusi thanks, also suggested by Kai. Now revert it